plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -639,12 +645,13 @@ func prepareMetricsReaderAndServer(t *testing.T, config config.Configuration, wa
 	mockPrometheus := startMockPrometheusServer(t, wantPromQlQuery, wantWarnings)
 
 	logger := zap.NewNop()
+	tracer := jtracer.NoOp()
```

Comment:
Looks like the tests are not testing tracing code. I would create a recording tracer here and add some validation in certain tests that the correct span is created.